macros/PGmatrixmacros.pl

```diff
@@ -775,11 +720,42 @@ =head4 create2d_matrix
 
 sub create2d_matrix {
 	my $string = shift;
-	my @rows = split("\\s*,\\s*",$string);
+	my @rows = split("\\s*[,;]\\s*",$string);
 	@rows = map {[split("\\s", $_ )]} @rows;
 	[@rows];
 }
 
+
+=head2 convert_to_array_ref {
+
+	$output_matrix = convert_to_array_ref($input_matrix)
+
+Converts a MathObject matrix (ref($input_matrix eq 'Value::Matrix')
+or a MatrixReal1 matrix (ref($input_matrix eq 'Matrix')to
+a reference to an array (e.g [[4,6],[3,2]]).
+This adaptor allows all of the LinearProgramming.pl subroutines to be used with
+MathObject arrays.
+
+$mathobject_matrix->value outputs an array (usually an array of array references) so placing it inside
+square bracket produces and array reference (of array references) which is what lp_display_mm() is
+seeking.
+
+=cut
+
+sub convert_to_array_ref {
+	my $input = shift;
+	if (ref($input) eq 'Value::Matrix' ) {
```

Comment:
This won't catch subclasses of `Value::Matrix`, so perhaps `if (Value::classMatch($input,"Matrix") {` would be better (this will return false if `$input` is not a MathObject at all, so should be safe).

Reply:
Accepted change.  Thanks.